include/sparrow/buffer.hpp

```diff
@@ -15,53 +15,172 @@
 #pragma once
 
 #include <algorithm>
+#include <cassert>
 #include <concepts>
 #include <cstdint>
 
+#include "sparrow/iterator.hpp"
+
 namespace sparrow
 {
     namespace impl
     {
-        template <class T>
-        struct buffer_data
+        // Should probably move to a MP util file
+        template <class T, bool is_const>
+        struct constify
+            : std::conditional<is_const, const T, T>
         {
-            using value_type = T;
-            using pointer = T*;
-            using size_type = std::size_t;
+        };
```

Comment:
Would you use it for all the implementations?

Reply:
I don't know yet for sure, thus the "probably" in the comment above.